File: Fugazi/src/org/fugazi/ast/Expression/literal/STRING.java
package org.fugazi.ast.expression.literal;

import org.fugazi.ast.expression.IExpressionVisitor;

public class STRING extends Literal {

    private final String value;

    public STRING(String _value) {
        this.value = _value;
    }

    public String getValue() {
        return this.value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    public <T> T accept(IExpressionVisitor<T> visitor) {
        return visitor.visitSTRING(this);
    }
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Literals/String.cs

namespace AST.Nodes.Literals
{
    public class String : Literal
    {
        private string value;
        public String(string value, PositionInText positionInText)
            : base(positionInText)
        {
            this.value = value;
        }
        public string GetValue()
        {
            return value;
        }
        
        public override string ToString()
        {
            return "string";
        }

        public override Types.Type RetrieveType()
        {
            return new Types.StringType();
        }

        public override T Accept<T>(VisitorInterfaces.IExpressionVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }
    }
}
